### DISCUSSION TITLE
GoPro Hero 12 GoPro Lab Hindsight not working

### DISCUSSION THREAD

**Discussion by FlyTieGuy** [gopro/labs#557] [General]
GoPro Labs on the hero 12.
The code for hindsight !MHSTO=60 worked on my GoPro 11 Black
But on the 12 it stops working again after 15 minutes, when I set to 60 minutes. It’s as if the code hasn’t registered at all. On the 11 when the camera is activated there is a message which states Features enabled and the code for Hindsight appears but that isn’t happening on the 12. This is an important feature for me so I’d appreciate a resolution.

**Comment by dnewman-gpsw** [comment]
It was missed in the port to HERO12.  It will be restored in a future Labs build.

**Reply by FlyTieGuy** [reply]
Thanks. That’s helpful. How often to builds typically take place

**Reply by dnewman-gpsw** [reply]
I expect this one will be about 6-8 weeks away.

**Reply by FlyTieGuy** [reply]
Will the latest Gopro Hero 12 update sort this one?

**Comment by esiremalaysia** [comment]
Is this issue correlated with #322 ? Because in my case I only use hs1 and hs0 to turn on/off the hindsight 15s window which is briefly working and then stops on the H12. Maybe that issue will be solved on future update, need more information about that.

**Reply by dnewman-gpsw** [reply]
It is not related.

**Comment by FlyTieGuy** [comment]
Hi will the latest Gopro Hero 12 firmware update take care of this?

**Reply by dnewman-gpsw** [reply]
No, the Labs (likely later today) will not contain this fix as the error report arrived after the QA had already begun.  It will be in the next Labs release in a month or so.

**Reply by FlyTieGuy** [reply]
Ok. You’re very good at keeping us up to date. Thanks

**Comment by ImmortlF350** [comment]
Thanks, appreciate folks looking into this for the next update.  I use the feature as well, for fishing, and my last outing, I was confused why some videos had the previous 30s and other did not.  

Not sure if this helps, but the behavior on my Hero 12 is a tick different than described above.  I DO see the HSTO=1440 when booting.  However after a period of time (believe about 15 min which would make sense), the actual recording of the loop seems to stop but the hindsight button is active.  If I wake up the screen by touching it after no camera interaction for 16 min or so,  the hindsight icon remains active and the counter starts indicating recording (+1s, +2s, +3s, etc...)  now that I woke it up,.  It seems to suddenly start recording the 30s loop when I wake up the screen. 

Thanks!

**Reply by FlyTieGuy** [reply]
Yes my understanding is that if you keep waking up the screen hindsight will continue to activate the problem is remembering to do that in my case!

**Comment by FlyTieGuy** [comment]
Hi has the labs Hindsight feature been reinstated to the firmware yet?

**Reply by dnewman-gpsw** [reply]
Check the release notes for your camera: https://gopro.github.io/labs/control/notes/

**Comment by FlyTieGuy** [comment]
Thanks it’s sorted for GoPro 12

**Comment by FlyTieGuy** [comment]
Just one more question on this. Is there any way of having Hindsight start automatically on power on?

**Reply by dnewman-gpsw** [reply]
You can auto start it with a booth command.  e.g. !MBOOT="!Lboot"!SAVEboot=hS1

**Reply by FlyTieGuy** [reply]
Do I need to split that into 2 sequences?  If I insert it all together the camera reads it and beeps but when I turn off and on again Hindsight has not activated

**Reply by dnewman-gpsw** [reply]
The command works as is when testing here.  It will start hindsight immediately. And it will start it at boot time.  It will not turn off the camera.

You can separate as two scans if you don't want to activate immediately.

!MBOOT="!Lboot"
!SAVEboot=hS1

If something else is happening, check your log file in MISC/qrlog.txt

**Reply by Ducnguyentq** [reply]
Hello, I applied !MBOOT="!Lboot"!SAVEboot=hS1 and it booted with gopro in 15 seconds. I want it to start with hindsight in 30 seconds. Can you assist me?

**Reply by FlyTieGuy** [reply]
Start with=hs2 instead of =hs1

**Reply by Ducnguyentq** [reply]
> !MBOOT="!Lboot"!SAVEboot=hS1

Nice . Thanks

**Reply by Ducnguyentq** [reply]
hi 
After inserting the battery, does the Gopro start up immediately? I want to turn it off and start it when I press the button
And I can't plug in the power bank for my gopro
Thanks

**Reply by FlyTieGuy** [reply]
When you insert the battery it should not power up. If you need to plug in your power bank to the GoPro  USB Pass Thru door

**Comment by Shane-william** [comment]
Hey, i added !MHSTO=1440 to my hero 12 and it displays when i boot but it still times out and shuts off at 15 mins sometimes even lass it seems with hindsight activated. Just wondering if anyone can help please

**Reply by dnewman-gpsw** [reply]
Make sure you are running the latest firmware as HERO12 with HSTO set to greater than 15 minutes is working fine in my tests.

**Reply by FlyTieGuy** [reply]
Still working for me too although do double check that it’s set to 1440

**Reply by Shane-william** [reply]
Thanks for taking the time to reply , I think i am running the latest firmware it says (02.10.70)

i am pretty new to gopro in general let alone labs ,  i followed directions from a gopro form , i put labs on the go pro , and the directions were as follows 

(Enter: !MHSTO=60 in the "additional commands" box. (60 min, 120, 1440 for 24 hours.) Note that is an exclamation point as the first character.

Save the generated QR Code. After installing labs simply point your camera at the QR code and it will automatically be read and installed. At this point I do not know if you need to read the code each time the camera is turned off/on. I'm going to read the code before a session just to make sure)

I entered       !MHSTO=1440
as to my understanding that changes the timeout to 24hrs  maybe i am missing Somthing as i never used labs before  

when i turn the gopro on it says (feature activated hsto=1440) but it is still timing out and now at about 5 mins i notice 🤦🏻‍♂️ , i will send a picture how i did the qr code if i can figure it out lol

**Comment by Shane-william** [comment]
[
![IMG_7144](https://github.com/gopro/labs/assets/163490979/02c93e4c-5058-4430-8004-890d512ba214)
](url)
![IMG_7145](https://github.com/gopro/labs/assets/163490979/af2d01e6-79d3-481f-acca-8f3daa525afa)

**Reply by FlyTieGuy** [reply]
Is auto power off or sleep set to never?

**Reply by Shane-william** [reply]
Yes sir it was and now is working perfect with that setting turned off. Im a bone head lololololol thank you both so Much for replying to me

**Reply by FlyTieGuy** [reply]
Glad you’re sorted. It’s a great option

**Comment by FlyTieGuy** [comment]
Hi All. Just wanted to check if the new HERO12 Black v2.20 firmware is compatible with the GOPRo Lab Hindsight function. I dont want to apply the update if I lose that feature. Many thanks

**Reply by dnewman-gpsw** [reply]
It is there.

**Reply by FlyTieGuy** [reply]
Many thanks

**Comment by FlyTieGuy** [comment]
Hi. Thinking of changing to GoPro 13. Will labs hindsight port over from release?  Also with voice control camera won’t turn off when hindsight is activated through labs. Is that to be expected?

**Reply by dnewman-gpsw** [reply]
Labs doesn't change voice control, are you sure?

**Comment by FlyTieGuy** [comment]
Yes I’m sure. When Hindsight is running GoPro start recording works fine as does GoPro Stop Recording but GoPro Turn off doesn’t. If Hindsight is off the Turnoff command works fine. I wondered whether it’s because the auto power off is set to never?

**Reply by dnewman-gpsw** [reply]
Is that the default firmware behavior?

**Comment by FlyTieGuy** [comment]
Also see first part of question above re Hero 13 and Labs Hindsight

**Reply by dnewman-gpsw** [reply]
Labs doesn't alter Hindsight, other than adding the 5 second option, otherwise it should behave like stock firmware.

**Reply by FlyTieGuy** [reply]
Will try it out again. Is labs hindsight working with hero 13?

**Reply by dnewman-gpsw** [reply]
This is no Labs variation of Hindsight, just stock Hindsight, which works on HERO13.

**Comment by FlyTieGuy** [comment]
I have just got the Mission 1 Pro and loaded Labs with the above codes for hindsight however hindsight isn’t turning on automatically after power up as it does with GP13. Do I need to make any changes

**Reply by dnewman-gpsw** [reply]
Which QR Code command?

**Comment by FlyTieGuy** [comment]
Also when hindsight is set to 30 in the sport video setting it works ok until it’s stopped manually and then turned back on using the shortcut key it reverts to 15seconds

**Reply by dnewman-gpsw** [reply]
I don't know what you mean.  Is this a Labs question, which Labs feature?

**Reply by FlyTieGuy** [reply]
Sorry I will try to be clearer. This relates to labs. The commands I use for my Hero 13 are as follows. They set the camera up so that hindsight switches on from boot up then stays on without turning off. 

hS2!MHSTO=1440
!MBOOT="!Lboot"
!SAVEboot=hS2

Initially this didn’t work on my new Mission Pro 1 but if I take the battery out after turning off then boot up it works ok. Is there a workaround to stop the need to remove the battery. 
The second issue is that when I go into the settings and manually set hindsight to 30 seconds it switches on ok but if I stop it and then switch it on using the shortcut it reverts to 15s

**Reply by dnewman-gpsw** [reply]
BOOT only happen when the camera does a full boot, not just waking for standby state. Standby is for 10 minutes for power button off.  No work around, other than using a !O command.  I will look into adding a Labs control for the standby timer length.

**Reply by dnewman-gpsw** [reply]
> The second issue is that when I go into the settings and manually set hindsight to 30 seconds it switches on ok but if I stop it and then switch it on using the shortcut it reverts to 15s

This part I don't understand.

**Reply by FlyTieGuy** [reply]
Thanks. What’s the !O command and how does that work. Thanks again for your attention to this

**Reply by FlyTieGuy** [reply]
So let’s say I go into the Sport preset by pressing the small oblong tablet at the bottom of the screen. I select hindsight and set it to 30s. When I come out of that hindsight is operating normally and counts up to 30s. If I press the stop hindsight button and then press the hindsight shortcut which I’ve set to bottom left on screen then hindsight switches on but only counts to 15s

**Reply by dnewman-gpsw** [reply]
So not Labs feature. This is way I didn't understand.

**Reply by dnewman-gpsw** [reply]
> Thanks. What’s the !O command and how does that work. Thanks again for your attention to this

!O - Off 

it is in the docs: https://gopro.github.io/labs/control/tech/

**Reply by FlyTieGuy** [reply]
So not Labs feature. This is way I didn't understand

Ah you’re absolutely right. How do I raise this to be addressed then

**Reply by dnewman-gpsw** [reply]
To me this doesn't look like an error, it is designed to be shortcut the 15s Hindsight.  The 15s/30s control, is not a saved setting, both start 15 or 30s hindsight.  So you are requesting a new feature or behavior.  https://gopro.com/en/us/info/static/feedback

**Reply by FlyTieGuy** [reply]
Thank you so much
